final_task/rss_reader/rss_parser.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3.8
+
+import html
+from os import path
+
+from bs4 import BeautifulSoup
+import feedparser
+from jinja2 import Environment, FileSystemLoader
+import requests
+
+from rss_reader import rss_exceptions
+
+
+class RSSparser:
+    """
+    Parsed the RSS news from received link, extracts required amount of news and prints result
+    in human readable format.
+
+    Parameters:
+        args: dict: parsed arguments out of sys.argv;
+        logger: LOGGER: tracks events that occur during program execution.
+
+    Returns:
+        store_news: list of dictionaries with extracted info of required amount of parsed news;
+        output_txt_news method returns result in human readable format (filling the prepared).
+    """
+
+    def __init__(self, args, logger):
+        self.url = args.source
+        self.limit = args.limit
+        self.logger = logger
+
+        self.logger.info(f'Get RSS_url {self.url} and value = {self.limit} (limits amount of output news)')
+
+        self.check_the_connection(self.url, self.logger)
+        self.response = self.get_the_response(self.url, self.logger)
+
+        # Extract the news-site name converting it to the unicode
+        self.feed_name = html.unescape(self.response.feed.get('title', ''))
+        # Extract all news separately in one list
+
+        self.logger.info('Separate a news from the URL.')
+
+        self.news = self.response.entries
+
+        self.limit = self.check_limit_value(self.limit)
+        self.all_news = self.parse_feed()
+
+    @staticmethod
+    def check_the_connection(url, logger):
+        """
+        Check the internet connection
+        :param url: str
+        :param logger: logger
+        :return: None
+        """
+        try:
+            requests.get(url, timeout=1)
+            logger.info('Check the Internet connection.')
+        except (
+                requests.exceptions.ConnectionError,
+                requests.exceptions.Timeout,
+                requests.exceptions.HTTPError,
+        ) as error:
+            raise SystemExit(f'Error: {error}. URL {url} is unreachable.')
```

Comment:
It is better just to reraise current exception here.

Reply:
But the output of this exceptions is too long, that is why I decided to raise SystemExit for making the error message more compact

Reply:
There are some better workarounds for this: you can either change current exception message or raise new exception with the same type and new message

Reply:
thanks. I'll try to manage this problem

Reply:
there are point in the task: In case of any mistakes utility should print human-readable error explanation. Exception tracebacks in stdout are prohibited in final version of application

Reply:
So, I still can't get better way to resolve this occasion

Reply:
Just raise your custom exception, not `SystemExit` and handle it in the main function